src/prog_models/prognostics_model.py

```diff
@@ -1113,21 +1126,78 @@ def calc_error(self, times: List[float], inputs: List[InputContainer], outputs:
         """
         method = kwargs.get('method', 'MSE')
 
-        # Call appropriate error calculation method
-        if method.lower() == 'mse':
-            return calc_error.MSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'max_e':
-            return calc_error.MAX_E(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'rmse':
-            return calc_error.RMSE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mae':
-            return calc_error.MAE(self, times, inputs, outputs, **kwargs)
-        if method.lower() == 'mape':
-            return calc_error.MAPE(self, times, inputs, outputs, **kwargs)
-
-        # If we get here, method is not supported
-        raise ProgModelInputException(f"Error method '{method}' not supported")
+        method_map = {
+            'mse': calc_error.MSE,
+            'max_e': calc_error.MAX_E,
+            'rmse': calc_error.RMSE,
+            'mae': calc_error.MAE,
+            'mape': calc_error.MAPE
+        }
+
+        try:
+            method = method_map[method.lower()]
+        except KeyError:
+            # If we get here, method is not supported
+            raise ProgModelInputException(f"Error method '{method}' not supported")
+        
+        acceptable_types = {dict, tuple, np.ndarray, list, SimResult, LazySimResult}
+        types = {type(times), type(inputs), type(outputs)}
+
+        if not all(t in acceptable_types for t in types):
+            type_error = f"Types passed in must be from the following list: np.ndarray, list, SimResult, or LazySimResult. Current types"
+            type_error += f" at data location {_loc}" if _loc is not None else ""
+            type_error += f": times = {type(times).__name__}, inputs = {type(inputs).__name__}, and outputs = {type(outputs).__name__}"
+            raise TypeError(type_error)
+        if len(times) != len(inputs) or len(inputs) != len(outputs):
+            len_error = "Times, inputs, and outputs must all be the same length. Current lengths"
+            len_error += f" at data location ({_loc})" if _loc is not None else ""
+            len_error += f": times = {len(times)}, inputs = {len(inputs)}, outputs = {len(outputs)}"
+            raise ValueError(len_error)
+        if len(times) < 2:
+            # raise ValueError(f"Must Provide at least 2 data points. Currently only passing in {len(times)}")
+            less_2_error = "Must provide at least 2 data points for times, inputs, and outputs"
+            less_2_error += f" at data location ({_loc})." if _loc is not None else ""
+            raise ValueError(less_2_error)
     
+        x = kwargs.get('x0', self.initialize(inputs[0], outputs[0]))
+        dt = kwargs.get('dt', 1e-99)
+        stability_tol = kwargs.get('stability_tol', 0.95)
```

Comment:
I'd suggest moving lines 1162-1164 down, right above the checks of these values on 1181, and adding a comment, something like "Checking keyword arguments have appropriate values." It was hard to read in the order it is now.